### ISSUE TITLE
LinearMixerState times become NaN in certain cases

### ISSUE THREAD

**Issue by zornor90** [KybernetikGames/animancer#34]
# Environment

- Animancer Version Number: 4.3
- Animancer Pro or Lite: Pro
- Unity Version: 2019.3
- Platform: PC

# Description

In Animancer 4.3 (and 4.2 as well I believe), it's possible that a LinearStateMixer will eventually become broken where both states are have a time of NaN and the animations no longer play.

An example of a broken state:
![image](https://user-images.githubusercontent.com/29684022/77118238-5f029780-69f9-11ea-9413-bf96195f4ec2.png)

This happens on an entity running a script that switches between the two states depending on movement speed as follows:
```
if (Mathf.Approximately(_pathfinder.velocity.magnitude, 0)) {
    currentState.stateMixer.Parameter = 0;
    currentState.stateMixer.Speed = currentState.idleAnimation.speed;
} else {
    currentState.stateMixer.Parameter = Mathf.Clamp(_pathfinder.velocity.magnitude, 0, 1);
    currentState.stateMixer.Speed = currentState.moveAnimation.speed;
}
```

**Comment by KybernetikGames** [comment]
I did a quick test and couldn't replicate the issue.

Are those `speed` values constant or do they change at all? What values do they commonly have?

Does the issue still occur if you don't set the `Parameter` and just leave it at say 0.5, or if you don't set the speed?

What output do you get if you Right Click on the Mixer in the Inspector and use the `Details -> Log Details of Everything` function before and after the time becomes NaN?

**Comment by KybernetikGames** [comment]
I still have not been able to replicate this issue, but I have fixed something else with mixer speeds that may be relevant. Go to the `ApplySynchroniseChildren` method in MixerState.cs and simply change the first line to:

```
var deltaTime = AnimancerPlayable.DeltaTime * EffectiveSpeed;
```

Please let me know if that fixes your issue.

**Comment by KybernetikGames** [comment]
This is now fixed in [Animancer v4.4](https://kybernetik.com.au/animancer/docs/changes/animancer-v4-4).
